e2e/rook-test-framework/enums/rook_platform_types.go

```diff
@@ -0,0 +1,41 @@
+package enums
+
+import (
+	"errors"
+	"strings"
+)
+
+type RookPlatformType int
+
+const (
+	Kubernetes RookPlatformType = iota + 1
+	BareMetal
+	StandAlone
+	None
+)
+
+var platforms = [...]string{
+	"Kubernetes",
+	"BareMetal",
+	"StandAlone",
+	"None",
+}
+
+func (platform RookPlatformType) String() string {
+	return platforms[platform-1]
+}
+
+func GetRookPlatFormTypeFromString(name string) (RookPlatformType, error) {
+	switch {
+	case strings.EqualFold(name, Kubernetes.String()):
+		return Kubernetes, nil
+	case strings.EqualFold(name, BareMetal.String()):
+		return BareMetal, nil
+	case strings.EqualFold(name, StandAlone.String()):
+		return StandAlone, nil
+	case strings.EqualFold(name, None.String()):
+		return None, nil
+	default:
+		return None, errors.New("Unsupported Rook Platform Type: " + name)
```

Comment:
in general, `fmt.Errorf` is preferred to `errors.New`.  Not sure if it's in the official linting rules, but I do believe it is preferred.